src/interface/blas2/backend/default.hpp

```diff
@@ -137,8 +137,21 @@ typename sb_handle_t::event_t _trsv(
     sb_handle_t& sb_handle, index_t _N, container_t0 _mA, index_t _lda,
     container_t1 _vx, increment_t _incx,
     typename sb_handle_t::event_t _dependencies) {
-  return blas::internal::_trsv_impl<4, 2, uplo, trn, diag>(
-      sb_handle, _N, _mA, _lda, _vx, _incx, _dependencies);
+  const auto device = sb_handle.get_queue().get_device();
+  const std::string vendor =
+      device.template get_info<sycl::info::device::vendor>();
+  if (device.is_gpu()) {
+    if (vendor.find("Intel") == vendor.npos) {
+      return blas::internal::_trsv_impl<32, 4, uplo, trn, diag>(
+          sb_handle, _N, _mA, _lda, _vx, _incx, _dependencies);
+    } else {
+      std::cout << "Trsv operator currently not supported on Intel GPUs\n";
```

Comment:
I'm not a fan of printing issues to `std::cout` but perhaps that's the convention in portBLAS?

Reply:
I am not a huge fan too, but I wanted to warn eventual user of it and also I did not like the idea of an exception here. There is not a convention in portBLAS. I think this is the first time we encounter a similar situation, so if you have any suggestion feel free to post it.

Reply:
Can you explain why you don't like the idea of an exception here?

Reply:
As far as I know we don't throw any exception(a part from SYCL one) anywhere else that's a first concern. Secondly, do we throw and catch it here? At this point a message and a solution to not let everything crash is enough imho. Do we throw it here and catch it elsewhere? where? Do we want to catch it?
This kind of questions/ideas are the reason why I did not put an exception here.

Reply:
You are absolutely right @pgorlani , totally my fault. I will update them all to throw a `std::runtime_error` exceptions.

Reply:
Thank you both, addressed in a626b95be70f2506014dc335b4f1950d8db4b119